liboptv/src/orientation.c

```diff
@@ -141,20 +149,803 @@ double weighted_dumbbell_precision(vec2d** targets, int num_targs, int num_cams,
     int pt;
     double dtot = 0, len_err_tot = 0, dist;
     vec3d res[2], *res_current;
-    
+
     for (pt = 0; pt < num_targs; pt++) {
         res_current = &(res[pt % 2]);
-        dtot += point_position(targets[pt], num_cams, multimed_pars, cals, 
+        dtot += point_position(targets[pt], num_cams, multimed_pars, cals,
             *res_current);
-        
+
         if (pt % 2 == 1) {
             vec_subt(res[0], res[1], res[0]);
             dist = vec_norm(res[0]);
             len_err_tot += 1 - ((dist > db_length) ? (db_length/dist) : dist/db_length);
         }
-    } 
-    
+    }
+
     /* note half as many pairs as targets is assumed */
     return (dtot / num_targs + db_weight*len_err_tot/(0.5*num_targs));
 }
 
+/*
+    Calibration* cal_in  -      camera calibration object
+    control_par *cpar -     control parameters
+    int nfix	-	 # of object points
+    vec3d fix[]	-	 object point data obtained using read_calblock() function and
+                    represents the known calibration body 3D points
+    target pix[] -	image coordinates obtained from sortgrid(). The points
+    which are associated with fix[] have real pointer (=i), others have -999
+    orient_par flags - structure of all the flags of the parameters to be (un)changed, 
+        read from orient.par parameter file using read_orient_par(), defaults are zeros;
+    Output:
+    Calibration *cal_in - if the orientation routine converged, this structure
+    is updated, otherwise, returned untouched. The routine works on a copy of
+    the calibration structure, cal.
+*/
+
+
+void orient (Calibration* cal_in, control_par *cpar, int nfix, vec3d fix[],
+            target pix[], orient_par *flags) {
+    int  	i,j,n, itnum, stopflag, n_obs=0, maxsize;
+//     int  	useflag=0, ccflag=0, scxflag=0, sheflag=0, interfflag=0, xhflag=0, yhflag=0,
+//             k1flag=0, k2flag=0, k3flag=0, p1flag=0, p2flag=0;
+    int  	intx1, intx2, inty1, inty2;
+
+    double  ident[IDT], XPX[NPAR][NPAR], XPy[NPAR], beta[NPAR], omega=0;
+    double  sigma0, sigmabeta[NPAR];
+    double 	xp, yp, xpd, ypd, xc, yc, r, qq, p, sumP;
+
+    int dummy, multi, numbers;
+
+    FILE *par_file;
+    double al,be,ga,nGl,e1_x,e1_y,e1_z,e2_x,e2_y,e2_z,safety_x,safety_y,safety_z;
+    double *P, *y, *yh, *Xbeta, *resi, *pixnr;
+    vec3d glass_dir, tmp_vec, e1, e2, pos;
+
+    Calibration *cal;
+
+    /* delta in meters and in radians for derivatives */
+    double  dm = 0.00001,  drad = 0.0000001;
+
+
+    /* we need to preserve the calibration if the model does not converge */
+    cal = malloc (sizeof (Calibration));
+    memcpy(cal, cal_in, sizeof (Calibration));
+
+    /* memory allocation. if all points nfix participate, then the 
+    n_obs = 2*nfix and the final size of the matrix, together with 
+    identities will be nfix*2 + 10 or in our settings 
+    nfix*2+IDT */
+    
+    maxsize = nfix*2 + IDT;
+    
+    P = (double *) calloc(maxsize, sizeof(double));
```

Comment:
Would appreciates some comments here saying what each matrix means, or maybe a citation of the algorithm's source. Or actually, why not both.